engine/src/main/java/org/destinationsol/game/asteroid/FarAsteroid.java

```diff
@@ -49,7 +52,41 @@ public boolean shouldBeRemoved(SolGame game) {
 
     @Override
     public SolObject toObj(SolGame game) {
-        return game.getAsteroidBuilder().build(game, myPos, myTex, mySz, myAngle, myRotSpd, mySpd, myRemoveController);
+        Vector2 position = adjustDesiredPosition(game, myPos, mySz);
```

Comment:
After this line, `myPos` and the actual position of asteroid (position) are out of sync. Way better to modify the value of myPos, rather than create a new Vector2. Luckily for you, that's exactly what you've done (by mistake).